main.js

```diff
@@ -0,0 +1,479 @@
+
+The event is called by clicking on the UI element "Reset".
+*/
+
+function resetCanvas() {
+    context.reset();
+    colorChange();
+    drawingStates.push(canvas.toDataURL());
+};
+
+resetTool.addEventListener("click", resetCanvas);
+
+/* Canvas Fill Tool
+
+We define the function that handles the canvas fill with the current color that is displayed in the UI and is also set with the changeColor function.
+
+Note: This overwrites anything already drawn on the canvas!
+
+It then "pushes" a state in the drawingState Array.
+
+The event is called by clicking on the UI element "Fill".
+*/
+
+function fillCanvas() {
+    context.fillRect(0,0,canvas.width,canvas.height);
+    drawingStates.push(canvas.toDataURL());
+};
+
+fillTool.addEventListener("click", fillCanvas);
+
+/* Undo function
+
+Defining the function which handles the undo event.
+
+We are updating the check checkStateNumber to be equal with the drawingStateNumber (which is used later at the redo button 
+to determin whether the user drew after the undo button in order to deactivate the redo button);
+
+We are updating the redoStates defined at the start of the code so that we can have a state to redo to, if needed;
+
+We are checking if there are any drawStates to which we can undo to;
+
+We are clearing the canvas before doing the undo, as the state previously saved would overlap the present canvas;
+
+We are "popping" the undo state from the drawingStates Array;
+
+We are importing the previous state (-1) as an img at 0,0 coordinates;
+
+Finally we have the two buttons that handle the event (one is present on the UI and the other one is the combination of CTRL + "z").
+*/
+
+function undo() {
+    checkStateNumber = drawingStateNumber;
+    redoStates.push(canvas.toDataURL());
+    if (drawingStates.length > 0) {
+        context.clearRect(0, 0, canvas.width, canvas.height);
+        drawingStates.pop();
+        const img = new Image();
+        img.src = drawingStates[drawingStates.length - 1];
+        img.onload = () => {
+            context.drawImage(img, 0, 0);
+        };
+    };
+};
+
+window.addEventListener("keydown", (ev) => {
+    if (ev.key === "z" && ev.ctrlKey) {
+        undo();
+    };
+});
+
+undoButton.addEventListener("click", undo);
+
+/* Redo function
+
+Defining the function which handles the redo event.
+
+We "push" the actual drawing state to the drawingStates Array so that we have a state to undo to, if needed;
+
+We check if we have redoStates to Redo to (one gets added each time the Undo event is triggered);
+
+We are importing the previous state (-1) as an img at 0,0 coordinates, however this time around we clear the canvas inside the img;
+
+Finally, we are "popping" the latest Redo state from the redoStates Array.
+
+*/
+
+function redo() {
+    drawingStates.push(canvas.toDataURL());
+    if (redoStates.length > 0) {
+        const img = new Image();
+        img.src = redoStates[redoStates.length - 1];
+        img.onload = () => {
```

Comment:
Poți folosi `addEventListener`.

```suggestion
        img.addEventListener('load', () => {
```